docs/articles/documentation/test-api/selecting-page-elements/selectors.md

```diff
@@ -123,221 +131,306 @@ You can initialize a selector with any of these objects.
+        .expect(osCount).eql(3)
+        .expect(submitButtonExists).ok();
 });
 ```
 
-Note that snapshot methods and property getters exposed through selectors are asynchronous.
+Note that selector property getters are asynchronous.
 
-### Selector Timeout
+### Obtain Element State
 
-When a selector is called in test code, TestCafe waits for the target node to appear
-in the DOM within the *selector timeout*.
+Selectors and promises returned by selectors expose API to get state (size, position, classes, etc.) of matched element.
+See [DOM Node State](./dom-node-state.md). Note that these methods and property getters
+are asynchronous so you can obtain element's property from a browser in the following way:
 
-You can specify the selector timeout in test code by using the [timeout](selector-options.md#optionstimeout) option.
-To set this timeout when launching tests, pass it to the [runner.run](../../using-testcafe/programming-interface/runner.md#run)
-method if you use API or specify the [selector-timeout](../../using-testcafe/command-line-interface.md#--selector-timeout-ms) option
-if you run TestCafe from the command line.
+```js
+const headerText = await Selector('#header').textConent;
+```
 
-Within the selector timeout, the selector is executed over and over again, until it returns a
-DOM node or the timeout exceeds.
+For example:
 
-Note that you can additionally require that the node returned by the selector is visible.
-To do this, use the [visibilityCheck](selector-options.md#optionsvisibilitycheck) option.
+```js
 
-## One-Time Selection
+import { Selector } from 'testcafe';
 
-To create a selector and immediately execute it without saving it,
-use the `select` method of the [test controller](../test-code-structure.md#test-controller).
+fixture `My fixture`
+    .page('http://devexpress.github.io/testcafe/example/');
 
-```text
-t.select( init [, options] )
-```
+const windowsInput = Selector('#windows');
 
-Parameter              | Type     | Description
----------------------- | -------- | ------------------------------------------------------------------
-`init`                 | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies a DOM node to be selected. See [Selector Initializers](#selector-initializers).
-`options`&#160;*(optional)* | Object   | See [Options](selector-options.md).
+test('Obtain Element State', async t => {
+    await t.click(windowsInput);
 
-The following example shows how to get a DOM element by ID with `t.select`.
+    const windowsInputChecked = await windowsInput.checked; // returns true
+});
 
-```js
-fixture `My fixture`
-    .page `http://www.example.com/`;
+ ```
 
-test('My Test', async t => {
-    const header = await t.select(() => document.getElementById('header'));
-});
-```
+#### DOM Node Snapshot
 
-## Using Selectors to Define Action Targets
+A selector allows to get a server-side representation of the DOM node's state from the webpage - *DOM node snapshot*.
```

Comment:
[Kazakov: rewrite to:]
A selector allows you to get a server-side representation of a DOM node's state from the tested webpage - *DOM node snapshot*.